models/image.py

```diff
@@ -1720,6 +1758,65 @@ def score(self):
     def score(self, value):
         self._score = value
 
+    @property
+    def psfflux(self):
+        """
+        The flux of the PSF (2D float array).
+        """
+        if self._data is None and self.filepath is not None:
+            self.load()
+        return self._psfflux
+
+    @psfflux.setter
+    def psfflux(self, value):
+        self._psfflux = value
+
+    @property
+    def psffluxerr(self):
+        """
+        The error on the flux of the PSF (2D float array).
+        """
+        if self._data is None and self.filepath is not None:
+            self.load()
+        return self._psffluxerr
+
+    @psffluxerr.setter
+    def psffluxerr(self, value):
+        self._psffluxerr = value
+
+    @property
+    def nandata(self):
+        """The image data, only masked with NaNs wherever the flag is not zero. """
+        if self._nandata is None:
+            self._nandata = self.data.copy()
+            if self.flags is not None:
+                self._nandata[self.flags != 0] = np.nan
```

Comment:
Ah, OK, now I see what nandata is.

Reply:
If we ever feel the need to preserve memory, we can just do imageobj.nandata = None, but I wonder if it's worth making a less "look at the internals of the object" interface for that.  (Not worth thinking about too hard right now.)

(A perlmutter CPU node has 128 cores (256 threads) and 512GB of memory.  I wouldn't run more than 128 processes at once, so each thread has 4GB of memory, so for the most part we shouldn't be in too much trouble.  A 2048x4096 image uses 32MB of memory if float, 64MB if double, so we can store 64 or 128 images before we've used up our memory.  Each image has something like 4 images (more if it's a subtraction), so we should be reasonably comfortable *except* if we try to keep everything in memory when doing a coadd.)

Reply:
We can also use `np.where()` to produce this on the fly. I'm not sure if the extra memory or the extra CPU cycles is more expensive for us.